src/modules/netinstall/PackageModel.h

```diff
@@ -0,0 +1,64 @@
+/* === This file is part of Calamares - <http://github.com/calamares> ===
+ *
+ *   Copyright (c) 2017, Kyle Robbertze <kyle@aims.ac.za>
+ *
+ *   Calamares is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation, either version 3 of the License, or
+ *   (at your option) any later version.
+ *
+ *   Calamares is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with Calamares. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef PACKAGEMODEL_H
+#define PACKAGEMODEL_H
+
+#include "PackageTreeItem.h"
+
+#include <QAbstractItemModel>
+#include <QObject>
+#include <QString>
+
+#include <yaml-cpp/yaml.h>
+
+// Required forward declarations
+class PackageTreeItem;
+
+class PackageModel : public QAbstractItemModel
+{
+    Q_OBJECT
+
+public:
+    explicit PackageModel( const YAML::Node& data, const QVariantList& columnHeadings,
+                           QObject* parent = 0 );
+    ~PackageModel();
+
+    QVariant data( const QModelIndex& index, int role ) const Q_DECL_OVERRIDE;
```

Comment:
It's ok to use plain `override` instead of `Q_DECL_OVERRIDE` because Calamares depends on C++11 and Qt 5.